website/docs/r/servicecatalog_product.html.markdown

```diff
@@ -0,0 +1,130 @@
+It is necessary to create an `aws_servicecatalog_portfolio_product_association`
+to link this to a portfolio before it can be provisioned,
+and for that portfolio to have an `aws_servicecatalog_portfolio_principal_association` explicitly
+opting-in those principals (users, groups, roles) which can access it.
+It is also common but not necessary to create launch constraints and additional provisioning artifacts
+to govern the usage of this Product.
+
+## Example Usage
+
+A rich example assuming the JSON is in a bucket already:
+
+```hcl
+resource "aws_servicecatalog_product" "test_existing_template" {
+  description         = "arbitrary product description"
+  distributor         = "arbitrary distributor"
+  name                = "My Existing Product"
+  owner               = "arbitrary owner"
+  product_type        = "CLOUD_FORMATION_TEMPLATE"
+  support_description = "arbitrary support description"
+  support_email       = "arbitrary@email.com"
+  support_url         = "http://arbitrary_url/foo.html"
+
+  provisioning_artifact {
+    description = "arbitrary description"
+    name        = "v1.0.0"
+    info {
+      LoadTemplateFromURL = "https://s3.amazonaws.com/bucket-xyz/cloudformation.json"
+    }
+  }
+
+  tags {
+    Foo = "bar"
+  }
+}
+```
+
+A minimal product example which also uploads a template to a bucket:
+
+```hcl
+resource "aws_s3_bucket" "my_product_template_bucket" {
+  bucket        = "%s"
+  region        = "us-east-1"
+  acl           = "private"
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_object" "my_product_template_object" {
+  bucket  = aws_s3_bucket.my_product_template_bucket.id
+  key     = "my_product_template.json"
+  content = <<EOF
+{
+  "AWSTemplateFormatVersion": "2010-09-09",
+  "Description": "Test CF teamplate for Service Catalog terraform dev",
+  "Resources": {
+    "Empty": {
+      "Type": "AWS::CloudFormation::WaitConditionHandle"
+    }
+  }
+}
+EOF
+}
+
+resource "aws_servicecatalog_product" "minimal_test_new_template" {
+  name         = "My New Minimal Product"
+  owner        = "arbitrary owner"
+  product_type = "CLOUD_FORMATION_TEMPLATE"
+
+  provisioning_artifact {
+    name = "v1.0.0"
+    info {
+      LoadTemplateFromURL = "https://s3.amazonaws.com/${aws_s3_bucket.my_product_template_bucket.id}/${aws_s3_bucket_object.my_product_template_object.key}"
+    }
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `description` - (Optional) The description of the product.
+* `distributor` - (Optional) The distributor of the product.
+* `name` - (Required) The name of the product.
+* `owner` - (Required) The owner of the product.
+* `product_type` - (Required) The type of product. Valid values: `CLOUD_FORMATION_TEMPLATE` or `MARKETPLACE`
+* `provisioning_artifact` - (Required) The configuration of the provisioning artifact. This object supports the following:
```

Comment:
lets break down config blocks to a separate doc block, see `cognito_user_pool` for example. it will make reading easier.